### PR TITLE
Update docs: Plambda, Plet, Pget and Prewrite

### PR DESCRIPTION
**Pull request by madskjeldgaard**
Hi everyone
I haven't done a PR for the SC source in a while so I hope I'm doing things correctly. 
Just a few updates to the docs:
## Purpose and Motivation

The help files for Plambda and Prewrite were a bit sparse and the examples a bit advanced, so I added some simple ones and cleaned up the text a bit. Plet and Pget did not have help files at all. 

## Types of changes

- Documentation

## To-do list

- [ ] Code is tested
- [ ] All tests are passing
- [ x] Updated documentation
- [ x] This PR is ready for review

### CODE REVIEW

**Comment by madskjeldgaard** [comment]
My commit message style for this PR was incredibly inconsistent, sorry about that

File: HelpSource/Classes/Plambda.schelp [hunk 1]
```diff
@@ -3,6 +3,10 @@ summary:: create a scope (namespace) for enclosed streams
 related:: Classes/Penvir, Classes/Pkey
 categories:: Streams-Patterns-Events>Patterns>Data Sharing
 
+Plambda is used to create a namespace for patterns to share data among eachother.
```
Comment (telephon) [inline HelpSource/Classes/Plambda.schelp]: typo: eachother –> each other

Comment (madskjeldgaard) [reply HelpSource/Classes/Plambda.schelp]: Thanks

**Comment by madskjeldgaard** [comment]
Added a help file for Ptrace as well

File: HelpSource/Classes/Ptrace.schelp [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+code::
+(
+// An event pattern playing random scale degrees
+p = Pbind(\dur, 0.125, \degree, Pwhite(0,10));
+
+// Se what the value of the scale degrees are using trace
+Ptrace(p, \degree, prefix: "Current scale degree: ").play
+)
+::
```
Comment (telephon) [inline HelpSource/Classes/Ptrace.schelp]: you could add a newline here for unix compatibility.

Comment (madskjeldgaard) [reply HelpSource/Classes/Ptrace.schelp]: In the prefix string do you mean or after the Ptrace line?

Comment (telephon) [reply HelpSource/Classes/Ptrace.schelp]: line 56

Comment (telephon) [reply HelpSource/Classes/Ptrace.schelp]: I meant just add a newline to the end of the file.

https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline

not a big thing, but still good to know.

Comment (madskjeldgaard) [reply HelpSource/Classes/Ptrace.schelp]: Thanks I did not know about this! I think I fixed it now

**Comment by madskjeldgaard** [comment]
> thanks! the contents look good but your git work here is messy. can you please update your branch so that it doesn't contain merge commits (the first two commits)? i would also recommend doing your work on a branch that isn't develop, as this is going to cause issues in repository management for you. if you'd like to resubmit this on a separate branch, feel free.
> 
> i don't normally do things the way you've done them in this PR, and i don't know the state of your local repo so it would be hard to give a surefire set of commands to run. but my best guess is:
> 
> ```shell
> git checkout develop
> git checkout -b topic/update-pdocs
> git rebase -i --onto upstream/develop develop
> # at this point, remove the two merge commits in the interactive rebase editor, then close the file
> git push -u origin topic/update-pdocs
> # and make a new pull-request from there
> ```
> 
> ```shell
> # assuming 'upstream' points to supercollider/supercollider,
> # you may also want to resync your local fork with upstream:
> git checkout develop
> git fetch upstream
> git reset --hard upstream/develop # WARNING - i take no responsibility for the result of this command!
> ```
> 
> see also https://github.com/supercollider/supercollider/wiki/git-workflow-and-guidelines for more info

Oh boy, yes I did mess up the git stuff here. I will look at your suggestions and try to fix it. Thanks!

**Comment by madskjeldgaard** [comment]
Sorry for the confusion but `git rebase -i --onto upstream/develop develop` gives me an empty interactive window it seems: 

(after making and switching to the new branch)

rebase

**Comment by brianlheim** [comment]
@madskjeldgaard sorry, but you are going to have to figure this one out on your own, like i said i don't know the state of your local repo + branches and i'd rather not make any more guesses. in the worst case you can create a new branch off upstream/develop and cherry-pick all the commits there.

**Comment by madskjeldgaard** [comment]
Alright, I will do that. I could do with some git exercise anyway




Get Outlook for Android







On Tue, Dec 24, 2019 at 2:52 AM +0100, "Brian Heim" <notifications@github.com> wrote:












@madskjeldgaard sorry, but you are going to have to figure this one out on your own, like i said i don't know the state of your local repo + branches and i'd rather not make any more guesses. in the worst case you can create a new branch off upstream/develop and cherry-pick all the commits there.

**Pull request closed without merging**